parsers/BR.py

```diff
@@ -1,191 +1,184 @@
-           'BR-S': u'sul'
-           }
-
-region_exchanges = {
-                    'BR-CS->BR-S': "sul_sudeste",
-                    'BR-CS->BR-NE': "sudeste_nordeste",
-                    'BR-CS->BR-N': "sudeste_norteFic",
-                    'BR-N->BR-NE': "norteFic_nordeste"
-                    }
-
-
-region_exchanges_directions = {
-                    'BR-CS->BR-S': -1,
-                    'BR-CS->BR-NE': 1,
-                    'BR-CS->BR-N': 1,
-                    'BR-N->BR-NE': 1
-                    }
-
-countries_exchange = {
-    'UY': {
-        'name': u'uruguai',
-        'flow': 1
-    },
-    'AR': {
-        'name': u'argentina',
-        'flow': -1
-    },
-    'PY': {
-        'name': u'paraguai',
-        'flow': -1
-    }
+URL = "http://tr.ons.org.br/Content/GetBalancoEnergetico/null"
+SOURCE = "ons.org.br"
+
+GENERATION_MAPPINGE = {
+    "nuclear": "nuclear",
+    "eolica": "wind",
+    "termica": "unknown",
+    "solar": "solar",
+    "hydro": "hydro",
+}
+
+REGIONS = {
+    "BR-NE": "nordeste",
+    "BR-N": "norte",
+    "BR-CS": "sudesteECentroOeste",
+    "BR-S": "sul",
+}
+
+REGION_EXCHANGES = {
+    "BR-CS->BR-S": "sul_sudeste",
+    "BR-CS->BR-NE": "sudeste_nordeste",
+    "BR-CS->BR-N": "sudeste_norteFic",
+    "BR-N->BR-NE": "norteFic_nordeste",
+}
+
+
+REGION_EXCHANGES_DIRECTIONS = {
+    "BR-CS->BR-S": -1,
+    "BR-CS->BR-NE": 1,
+    "BR-CS->BR-N": 1,
+    "BR-N->BR-NE": 1,
+}
+
+COUNTRIES_EXCHANGE = {
+    "UY": {"name": "uruguai", "flow": 1},
+    "AR": {"name": "argentina", "flow": -1},
+    "PY": {"name": "paraguai", "flow": -1},
 }
 
 
 def get_data(session, logger):
     """Requests generation data in json format."""
-
     s = session or requests.session()
-    json_data = s.get(url).json()
+    json_data = s.get(URL).json()
+
     return json_data
 
 
 def production_processor(json_data, zone_key) -> tuple:
     """Extracts data timestamp and sums regional data into totals by key."""
 
-    dt = arrow.get(json_data['Data'])
+    dt = arrow.get(json_data["Data"])
     totals = defaultdict(lambda: 0.0)
 
-    region = regions[zone_key]
-    breakdown = json_data[region][u'geracao']
+    region = REGIONS[zone_key]
+    breakdown = json_data[region]["geracao"]
     for generation, val in breakdown.items():
         totals[generation] += val
 
     # BR_CS contains the Itaipu Dam.
     # We merge the hydro keys into one, then remove unnecessary keys.
-    totals['hydro'] = totals.get(u'hidraulica', 0.0) + totals.get(u'itaipu50HzBrasil', 0.0) + totals.get(u'itaipu60Hz', 0.0)
-    entriesToRemove = (u'hidraulica', u'itaipu50HzBrasil', u'itaipu60Hz', u'total')
-    for k in entriesToRemove:
+    totals["hydro"] = (
+        totals.get("hidraulica", 0.0)
+        + totals.get("itaipu50HzBrasil", 0.0)
+        + totals.get("itaipu60Hz", 0.0)
+    )
+    entries_to_remove = ("hidraulica", "itaipu50HzBrasil", "itaipu60Hz", "total")
+    for k in entries_to_remove:
         totals.pop(k, None)
```

Comment:
Refactor to be more Pythonic, using a set and a dict comprehension

```suggestion
    entries_to_remove = {"hidraulica", "itaipu50HzBrasil", "itaipu60Hz", "total"}
    totals = {k, v for k, v in totals.items() if k not in entries_to_remove}
```